### ISSUE TITLE
constructor error, html not decoded, fix included

### ISSUE THREAD

**Issue by bendichter** [pytube/pytube#72]
I found the following bug:

``` python
from pytube import YouTube
from pprint import pprint
yt = YouTube("http://www.youtube.com/watch?v=Ik-RsDGPI5Y")

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-26614df866ad> in <module>()
----> 1 yt = YouTube("http://www.youtube.com/watch?v=Ik-RsDGPI5Y")

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in __init__(self, url)
     69         self._videos = []
     70         if url:
---> 71             self.from_url(url)
     72 
     73     @property

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in from_url(self, url)
    162 
    163         # Get the video details.
--> 164         video_data = self.get_video_data()
    165 
    166         # Set the title from the title.

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in get_video_data(self)
    263 
    264         html = response.read()
--> 265         if "og:restrictions:age" in html:
    266             raise AgeRestricted("Age restricted video. Unable to download "
    267                                 "without being signed in.")

TypeError: 'str' does not support the buffer interface

In [5]: yt = YouTube('http://www.youtube.com/watch?v=Ik-RsDGPI5Y')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-0a20bae49aa2> in <module>()
----> 1 yt = YouTube('http://www.youtube.com/watch?v=Ik-RsDGPI5Y')

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in __init__(self, url)
     69         self._videos = []
     70         if url:
---> 71             self.from_url(url)
     72 
     73     @property

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in from_url(self, url)
    162 
    163         # Get the video details.
--> 164         video_data = self.get_video_data()
    165 
    166         # Set the title from the title.

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in get_video_data(self)
    263 
    264         html = response.read()
--> 265         if "og:restrictions:age" in html:
    266             raise AgeRestricted("Age restricted video. Unable to download "
    267                                 "without being signed in.")

TypeError: 'str' does not support the buffer interface

In [6]: yt = YouTube(u'http://www.youtube.com/watch?v=Ik-RsDGPI5Y')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-6-3ab2a932c403> in <module>()
----> 1 yt = YouTube(u'http://www.youtube.com/watch?v=Ik-RsDGPI5Y')

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in __init__(self, url)
     69         self._videos = []
     70         if url:
---> 71             self.from_url(url)
     72 
     73     @property

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in from_url(self, url)
    162 
    163         # Get the video details.
--> 164         video_data = self.get_video_data()
    165 
    166         # Set the title from the title.

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/pytube/api.py in get_video_data(self)
    263 
    264         html = response.read()
--> 265         if "og:restrictions:age" in html:
    266             raise AgeRestricted("Age restricted video. Unable to download "
    267                                 "without being signed in.")

TypeError: 'str' does not support the buffer interface
```

I fixed the bug by changing api.py line 264 from

``` python
html = response.read()
```

to

``` python
html = response.read().decode()
```

I think it may be an issue with py2->py3 compatibility. Anyway, I'm pretty sure that fixed it, though I don't know if it broke anything downstream.

**Comment by nficano** [comment]
@bendichter I just tried that but it breaks the unit tests. Thanks for reporting this I'll look into it more tonight.

**Comment by bendichter** [comment]
Yes, I suspected a proper correction might be a bit more involved. Let me know if I can help you reproduce the error.

**Comment by nficano** [comment]
@bendichter I couldn't reproduce this under Python 2.7... (see the traceback below). I'm going to add a test case for this now, but can you please verify you're running the latest version and let me know what version of python you're running? 

``` python
In [2]: from pytube import YouTube

In [3]: yt = YouTube('http://www.youtube.com/watch?v=Ik-RsDGPI5Y')

In [4]: yt = YouTube('https://www.youtube.com/watch?v=nzNgkc6t260')
---------------------------------------------------------------------------
AgeRestricted                             Traceback (most recent call last)
<ipython-input-4-c9756734f7d1> in <module>()
----> 1 yt = YouTube('https://www.youtube.com/watch?v=nzNgkc6t260')

/Users/nficano/Projects/pytube/pytube/api.py in __init__(self, url)
     63         self._videos = []
     64         if url:
---> 65             self.from_url(url)
     66
     67     @property

/Users/nficano/Projects/pytube/pytube/api.py in from_url(self, url)
    156
    157         # Get the video details.
--> 158         video_data = self.get_video_data()
    159
    160         # Set the title from the title.

/Users/nficano/Projects/pytube/pytube/api.py in get_video_data(self)
    258         html = response.read()
    259         if "og:restrictions:age" in html:
--> 260             raise AgeRestricted("Age restricted video. Unable to download "
    261                                 "without being signed in.")
    262

AgeRestricted: Age restricted video. Unable to download without being signed in.

In [5]:
```

**Comment by nficano** [comment]
Ok test added: https://github.com/nficano/pytube/blob/master/tests/test_pytube.py#L85-L96

**Comment by nficano** [comment]
I'm going to close this, please reopen if you have more details.

**Comment by bendichter** [comment]
I'm not at my computer but I'm pretty sure it was 3.5

**Comment by nficano** [comment]
My apologies, a patch for this has just been released (6.1.0) sorry for dismissing this as "not a bug", I assumed my tests were right without confirming it myself. lesson learned :)

**Comment by bendichter** [comment]
Oh no worries. Thanks for letting me know it's been fixed.
